index.js

```diff
@@ -195,4 +195,30 @@ form.addEventListener('submit', (event) => {
     event.preventDefault();
     errorMessage();
   }
-});
\ No newline at end of file
+});
+
+// local storage
+const nameInput = document.getElementById('name');
+const emailInput = document.getElementById('email');
+const msgInput = document.getElementById('message');
+
+function saveData() {
+  const userData = {
+    name: nameInput.value,
+    email: emailInput.value,
+    message: msgInput.value,
+
+  };
+  localStorage.setItem('userData', JSON.stringify(userData));
+}
+nameInput.addEventListener('focusout', saveData);
+emailInput.addEventListener('focusout', saveData);
+msgInput.addEventListener('focusout', saveData);
+
+const data = JSON.parse(localStorage.getItem('userData'));
+
+if (data) {
+  nameInput.value = data.name;
+  emailInput.value = data.email;
+  msgInput.value = data.message;
+}
```

Comment:
- [OPTIONAL] I think you should consider using the onload event listener to deteremine when to check localstorage for data to populate the form fields. The could look like below. What do you think ? 🤔 

```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```